### PR TITLE
docs: refine docs

### PR DESCRIPTION
**Pull request by Shigma**
- Updated some obsolete documents with new features
- Added some caveats related to recent issues
- Improved Chinese translation consistency
- Optimized `<Badge>` style
- Added and removed some badges
- Renamed custom container `vue` to `files`
- Adding whitespace between language and line numbers for better highlighting in IDEs

### CODE REVIEW

**Comment by Shigma** [comment]
See:

- packages/@vuepress/theme-default/global-components/Badge.vue
- packages/docs/docs/guide/markdown.md
- packages/docs/docs/zh/guide/markdown.md
- packages/docs/docs/plugin/official/plugin-clean-urls.md
- packages/docs/docs/zh/plugin/official/plugin-clean-urls.md

File: packages/docs/docs/zh/plugin/official/plugin-nprogress.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ metaTitle: Nprogress 插件 | VuePress
 
 # [@vuepress/plugin-nprogress](https://github.com/vuejs/vuepress/tree/master/packages/%40vuepress/plugin-nprogress)
 
-> Nprogress plugin
+一个基于 [nprogress](https://github.com/rstacruz/nprogress) 的进度条插件。<Badge text="1.0.0-alpha.31+" vertical="baseline"/>
```
Comment (ulivz) [inline packages/docs/docs/zh/plugin/official/plugin-nprogress.md]: I don't think that we should use `vertical-align: baseline ` here, and the default `top` is intentional.

File: packages/docs/docs/guide/basic-config.md [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
+::: files
 .
-├─ docs
-│  ├─ README.md
-│  └─ .vuepress
-│     └─ config.js
+├── docs
+│   ├── README.md
+│   └── .vuepress
+│       └── config.js
```
Comment (ulivz) [inline packages/docs/docs/guide/basic-config.md]: Don't think it's necessary to add indent here.

File: packages/docs/docs/guide/markdown.md [hunk 3]
```diff
@@ -8,44 +8,39 @@ Headers automatically get anchor links applied. Rendering of anchors can be conf
...
-```
+├── README.md
+├── foo
+│   ├── README.md
+│   ├── one.md
+│   └── two.md
+└── bar
+&nbsp;   ├── README.md
+&nbsp;   ├── three.md
+&nbsp;   └── four.md
```
Comment (ulivz) [inline packages/docs/docs/guide/markdown.md]: Same here, and please remove `&nbsp;` here.

BTW, I don't think `files` container should be used in any other location except for two parts of the directory structure.

File: packages/docs/docs/guide/permalinks.md [hunk 4]
```diff
@@ -4,14 +4,15 @@
...
-    ├── _post
-    │   └── intro-vuepress.md
-    ├── index.md
-    └── tags.md
-```
+&nbsp;   ├── _post
+&nbsp;   │   └── intro-vuepress.md
+&nbsp;   ├── index.md
+&nbsp;   └── tags.md
+:::
```
Comment (ulivz) [inline packages/docs/docs/guide/permalinks.md]: Same here.

File: packages/docs/docs/guide/using-vue.md [hunk 5]
```diff
@@ -98,15 +98,15 @@ By default, fenced code blocks are automatically wrapped with `v-pre`. If you wa
...
-      └─ Foo
-         └─ Bar.vue
-```
+└── .vuepress
+&nbsp;   └── components
+&nbsp;       ├── demo-1.vue
+&nbsp;       ├── OtherComponent.vue
+&nbsp;       └── Foo
+             └── Bar.vue
+:::
```
Comment (ulivz) [inline packages/docs/docs/guide/using-vue.md]: Same here.

**Comment by Shigma** [comment]
@ulivz I can of course remove these indents. I am doing these based on the consideration of unity. Since the other places took 4 indentations, I modified the indentation here.

Maybe I will all change to 3-indented.

File: packages/docs/docs/guide/markdown.md (hunk 3 above)
Comment (Shigma) [reply packages/docs/docs/guide/markdown.md]: There are actually more than two parts that show directory structure.

**Comment by Shigma** [comment]
Maybe I should separate this pr into several ones. It is too big now.

**Pull request closed without merging**